File: 2024-2/0682/unidad07.b - Programación distribuida con BD/SoftPub_Front-end/SoftPubPersonaBO/EspecialidadBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubBaseBO;
using SoftPubBaseBO.ServicioWeb;

namespace SoftPubPersonaBO
{
    public class EspecialidadBO : BaseBO
    {
        public int insertar(int idEspecialidad, string nombre_especialidad, string facultad, nivel _nivel)
        {
            return WsCliente.especialidad_insertar(idEspecialidad, nombre_especialidad, facultad, _nivel);
        }

        public int modificar(int idEspecialidad, string nombre_especialidad, string facultad, nivel _nivel)
        {
            return WsCliente.especialidad_modificar(idEspecialidad, nombre_especialidad, facultad, _nivel);    
        }

        public int eliminar(int idEspecialidad)
        {
            return WsCliente.especialidad_eliminar(idEspecialidad);    
        }

        public BindingList<especialidad> listarTodos()
        {
            especialidad[] arreglo = WsCliente.especialidad_listarTodos();
            return new BindingList<especialidad>(arreglo);
        }

        public especialidad obtenerPorId(int idEspecialidad)
        {
            return WsCliente.especialidad_obtenerPorId(idEspecialidad);
        }
    }
}

File: 2024-2/0682/unidad07.f - Programación distribuida - reportes/SoftPub_Back-end/ServidorRMI/SoftPubRMI_Servidor/src/pe/edu/pucp/softpub/rmi/interfaces/EspecialidadBO.java
package pe.edu.pucp.softpub.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.model.Especialidad;
import pe.edu.pucp.softpub.persona.model.Nivel;

public interface EspecialidadBO extends Remote {

    public Integer insertar(Integer idEspecialidad, String nombre_especialidad, String facultad, Nivel nivel) throws  RemoteException;

    public Integer modificar(Integer idEspecialidad, String nombre_especialidad, String facultad, Nivel nivel) throws  RemoteException;

    public Integer eliminar(Integer idEspecialidad) throws  RemoteException;

    public ArrayList<Especialidad> listarTodos() throws  RemoteException;

    public Especialidad obtenerPorId(Integer idEspecialidad) throws  RemoteException;
}
